lib/Dialect/CGGI/IR/CGGIOps.td

```diff
@@ -270,4 +273,33 @@ def CGGI_MultiLutLinCombOp : CGGI_Op<"multi_lut_lincomb", [
   let hasVerifier = 1;
 }
 
+def CGGI_SubOp : CGGI_Op<"sub", [
+    Pure,
+    SameOperandsAndResultType,
+    ElementwiseMappable,
+    Scalarizable
+]> {
+  let arguments = (ins LWECiphertextLike:$lhs, LWECiphertextLike:$rhs);
+  let results = (outs LWECiphertextLike:$output);
+  let assemblyFormat = "operands attr-dict `:` qualified(type($output))";
+  let summary = "Subtraction of two ciphertexts.";
+}
+
+
+def CGGI_ShiftRightOp : CGGI_Op<"shr", [
+    Pure,
+]> {
+  let arguments = (ins LWECiphertextLike:$lhs, AnyI8:$shiftAmount);
+  let results = (outs LWECiphertextLike:$output);
+  let summary = "Arithmetic shift to right of a ciphertexts by an integer.";
```

Comment:
Is this also just here to be part of the TFHE-rs API? (Why wouldn't one implement this as a scalar-ciphertext mul-by-2?)

Maybe document at least that it's here for tfhe-rs.

Reply:
I do not want to use it as a arithmetic division by two, more of a way to extract the MSB from a ciphertext